catch_analysis_tools/movingstack.py

```diff
@@ -0,0 +1,262 @@
+    
+    Parameters:
+    ra (float): Right Ascension (RA) of the reference position in degrees.
+    dec (float): Declination (Dec) of the reference position in degrees.
+    size_x (int): Number of pixels along the x-axis (image width).
+    size_y (int): Number of pixels along the y-axis (image height).
+    pixel_scale (float): Pixel scale in arcseconds per pixel (default is 0.4"/pixel).
+    
+    Returns:
+    wcs: An astropy WCS object.
+    """
+
+    # Create a WCS object
+    wcs = WCS(naxis=2)
+
+    # Set the reference pixel to the center of the image
+    wcs.wcs.crpix = [size_x / 2, size_y / 2]
+
+    # Set the reference position (CRVAL) in degrees
+    wcs.wcs.crval = [ra, dec]
+
+    # Set the pixel scale (CDELT) in degrees per pixel
+    #wcs.wcs.cdelt = np.array([-pixel_scale / 3600, pixel_scale / 3600])
+    wcs.wcs.cdelt = np.array([-pixel_scale, pixel_scale])
+
+    # Set the coordinate system to RA/Dec with degrees as units
+    wcs.wcs.ctype = ["RA---TAN", "DEC--TAN"]
+
+    return wcs
+
+def project_fits_to_wcs(input_fits, new_wcs, shape_out, photometric_cal=True, subtract=True, intermediate=False):
+    """
+    Project a FITS image onto a new WCS using the reproject library.
+
+    Parameters:
+    input_fits (str): Path to the input FITS file.
+    new_wcs (astropy.wcs.WCS): The target WCS object.
+    shape_out (tuple): The shape (height, width) of the output image.
+
+    Returns:
+    reprojected_data (2D numpy array): The image reprojected onto the new WCS.
+    """
+    # Open the input FITS file and extract the data and WCS
+    with fits.open(input_fits) as hdu_list:
+      input_data = hdu_list[0].data
+      header = hdu_list[0].header
+    original_wcs = get_wcs(input_fits)
+
+    # Photometric calibration
+    if (photometric_cal):
+      magzp = get_photozp(header)
+      input_data = input_data * 10**(-0.4 * (magzp - defaultzeropoint))
+      if (intermediate):
+        hdu=fits.PrimaryHDU(data=input_data,header=header)
+        hdu.writeto(os.getcwd()+"/"+os.path.splitext(os.path.basename(input_fits))[0]+".photocal.fit",overwrite=True)
+
+    # Subract background
+    if (subtract):
+      input_data = input_data.byteswap().newbyteorder()
+      try:
+        bkg = sep.Background(input_data)
+      except ValueError:
+        input_data = input_data.byteswap().newbyteorder()
+        bkg = sep.Background(input_data)
+      input_data = input_data - bkg
+      if (intermediate):
+        hdu=fits.PrimaryHDU(data=bkg,header=header)
+        hdu.writeto(os.getcwd()+"/"+os.path.splitext(os.path.basename(input_fits))[0]+".bkg.fit",overwrite=True)
+        hdu=fits.PrimaryHDU(data=input_data,header=header)
+        hdu.writeto(os.getcwd()+"/"+os.path.splitext(os.path.basename(input_fits))[0]+".subtracted.fit",overwrite=True)
+
+    # Reproject the image onto the new WCS
+    reprojected_data, _ = reproject_interp((input_data, original_wcs), new_wcs, shape_out=shape_out)
```

Comment:
Since the output coverage map isn't used, we can save some CPU cycles with `return_footprint=False`.